src/runtime/state.rs

```diff
@@ -0,0 +1,144 @@
+use crate::error::OciSpecError;
+
+use derive_builder::Builder;
+use getset::{Getters, MutGetters, Setters};
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, fmt::Display};
+
+/// ContainerState represents the state of a container.
+#[derive(Clone, Copy, Debug, Default, Eq, Hash, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum ContainerState {
+    /// Creating indicates that the container is being created,
+    Creating,
+
+    /// Created indicates that the runtime has finished the create operation,
+    /// but the container exists but has not been run yet.
+    Created,
+
+    /// Running indicates that the container process has executed the
+    /// user-specified program but has not exited
+    Running,
+
+    /// Paused indicates that the container exists, but all processes are paused.
+    Paused,
```

Comment:
Is this something official? I can't see it in https://github.com/opencontainers/runtime-spec/blob/main/specs-go/state.go

Reply:
I'm so sorry @saschagrunert
I think I misunderstood the State of things and mixed them up. I need to rethink it. Thank you for your reminder.